### PR TITLE
fix: modify [`setIsWormholeEvmChain`] to take bool

### LINKED ISSUE #250: modify evm chain id registration
a registered EVM compatible chain diverges as the result of an upgrade, the existing implementation of `WormholeTransceiverState::setIsWormholeEvmChain` means that it will not be possible to update the corresponding storage. If, for whatever reason, a chain such as Optimism decides to move away from EVM (kind of the opposite of what they did in reality, going from OVM to EVM), its chain ID will now correspond to a non-EVM chain.  This function should take a boolean argument,

### PR DESCRIPTION
**Pull request by RahulMaganti47**
Fixes #250.

### CODE REVIEW

File: evm/src/interfaces/IWormholeTransceiverState.sol [hunk 1]
```diff
@@ -50,7 +50,7 @@ interface IWormholeTransceiverState {
     /// @notice Set whether the chain is EVM compatible.
     /// @dev This function is only callable by the `owner`.
     /// @param chainId The Wormhole chain ID to set.
```
Comment (djb15) [inline evm/src/interfaces/IWormholeTransceiverState.sol:52]: Can you update the docstring to include the new param pls?

Comment (RahulMaganti47) [reply evm/src/interfaces/IWormholeTransceiverState.sol:52]: yea, fixed.

File: evm/test/IntegrationRelayer.t.sol [hunk 2]
```diff
@@ -172,7 +172,7 @@ contract TestEndToEndRelayer is
         nttManagerChain2.setPeer(chainId1, bytes32(uint256(uint160(address(nttManagerChain1)))), 9);
         DummyToken token2 = DummyTokenMintAndBurn(nttManagerChain2.token());
         wormholeTransceiverChain2.setIsWormholeRelayingEnabled(chainId1, true);
-        wormholeTransceiverChain2.setIsWormholeEvmChain(chainId1);
+        wormholeTransceiverChain2.setIsWormholeEvmChain(chainId1, true);
```
Comment (gator-boi) [inline evm/test/IntegrationRelayer.t.sol:175]: We should add a test demonstrating that we can set the `IsWormholeEvmChain` to false after setting it to true.

File: evm/test/IntegrationRelayer.t.sol [hunk 3]
```diff
@@ -201,8 +201,24 @@ contract TestEndToEndRelayer is
...
-
             bytes memory instructions = encodeTransceiverInstruction(false);
+
+            // set invalid config
+            vm.stopPrank();
+            wormholeTransceiverChain1.setIsWormholeEvmChain(chainId2, false);
+
+            // config not set correctly
+            vm.startPrank(userA);
+            vm.expectRevert();
```
Comment (djb15) [inline evm/test/IntegrationRelayer.t.sol:212]: Nit pick, but would be good to expect on a specific error

**Review by djb15 — APPROVED** [review]

**Review by gator-boi — APPROVED** [review]

**Review by djb15 — APPROVED** [review]

**Review by gator-boi — APPROVED** [review]

**Pull request merged**